lib/Report/SessionHistory.php

```diff
@@ -361,6 +361,87 @@ public function getResults(SanitizerInterface $sanitizedParams)
                 $rows[] = $logRecord;
             }
 
+            return new ReportResult(
+                $metadata,
+                $rows,
+                count($rows),
+            );
+        } else {
+            $params = [
+                'fromDt' => $fromDt->format(DateFormatHelper::getSystemFormat()),
+                'toDt' => $toDt->format(DateFormatHelper::getSystemFormat()),
+            ];
+
+            $sql = 'SELECT
+             `session_history`.`sessionId`,
+             `session_history`.`startTime`,
+             `session_history`.`userId`,
+             `session_history`.`userAgent`,
+             `session_history`.`ipAddress`,
+             `user`.`userName`,
+             `usertype`.`userType`,
+             `session`.`session_expiration`
+                FROM `session_history`
+                    LEFT OUTER JOIN `user` ON `user`.`userId` = `session_history`.`userId`
+                    LEFT OUTER JOIN `usertype` ON `usertype`.`userTypeId` = `user`.`userTypeId`
+                    LEFT OUTER JOIN `session` ON `session`.`session_data` 
+                        LIKE CONCAT("sessionHistoryId|i:", session_history.sessionId, "%")
```

Comment:
Needing to do this sort of join points to a data model problem, it also means that as the sessions are removed from the session table, we lose the `session_expiration` and therefore the end date and duration from the report.

Ideally we would have a `lastUsedTime` field in the session_history table which is updated each time an existing session is touched, and would then accurately represent the time that session was active.

That is outside the scope of the initial issue, but please you could investigate adding this column to the table and making sure it gets updated when the session is accessed?